### DISCUSSION TITLE
Prioritizing topics for TRAPI 1.1 by vote

### DISCUSSION THREAD

**Discussion by edeutsch** [NCATSTranslator/ReasonerAPI#213] [General]
This thread will serve as an experimental voting platform for prioritizing the topics to tackle next. After everyone votes, we will tackle them roughly in order of voting.

You are NOT voting for the PRs as written.
You ARE voting to prioritize the topics to be resolved and included for April 1 TRAPI 1.1 release!

Vlado's Voting Scheme:
+2 Highest (heart) - Must get this done for TRAPI 1.1 April 1st
+1 High (thumb up) - Should try to get this done for TRAPI 1.1 April 1st
0  Medium (none) - Meh, if there's extra time by April 1st, sure
-1 Low Priority (thumb down) - This should not be prioritized. TRAPI 1.2 is better.

Vote this comment if you like the system!

**Comment by edeutsch** [comment]
Migrate /predicates to /knowledge_map
- https://github.com/NCATSTranslator/ReasonerAPI/pull/171
- https://github.com/NCATSTranslator/ReasonerAPI/issues/143
- https://github.com/NCATSTranslator/ReasonerAPI/issues/142
- https://github.com/NCATSTranslator/ReasonerAPI/issues/156
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Reply by vdancik** [reply]
This topic is now represented by PR #215

**Comment by edeutsch** [comment]
What do category/predicate/id lists mean?
- https://github.com/NCATSTranslator/ReasonerAPI/issues/199
- https://github.com/NCATSTranslator/ReasonerAPI/issues/145
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Reply by patrickkwang** [reply]
This may be a subset of "context queries" [below](https://github.com/NCATSTranslator/ReasonerAPI/discussions/213#discussioncomment-359918).

**Comment by edeutsch** [comment]
Batch query support
- https://github.com/NCATSTranslator/ReasonerAPI/issues/145
- https://github.com/NCATSTranslator/ReasonerAPI/issues/124
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
(Q)Edge.relation should be an ontology CURIE
- #176
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Context Queries (where clause in the query, <, >, <=, >=, etc.)
- https://github.com/NCATSTranslator/ReasonerAPI/pull/157
- https://github.com/NCATSTranslator/ReasonerAPI/issues/104
- https://github.com/NCATSTranslator/ReasonerAPI/issues/206
- https://github.com/NCATSTranslator/ReasonerAPI/issues/162 ?
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Add 'score' property to EdgeBinding
- https://github.com/NCATSTranslator/ReasonerAPI/pull/181
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Reply by ehinderer** [reply]
Thumbs down because this is dependent on EPC decisions, not because it's not important.

**Comment by edeutsch** [comment]
Asynchronous query mechanisms (including callback)
- https://github.com/NCATSTranslator/ReasonerAPI/pull/147
- https://github.com/NCATSTranslator/ReasonerAPI/pull/150
- https://github.com/NCATSTranslator/ReasonerAPI/issues/96
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Support for operations{}
- https://github.com/NCATSTranslator/ReasonerAPI/pull/200
- https://github.com/NCATSTranslator/ReasonerAPI/issues/152
- this is still a draft
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Add mechanism for KPs to announce whether they perform node normalization
- https://github.com/NCATSTranslator/ReasonerAPI/issues/188
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Allow query to specify 'log level'
- https://github.com/NCATSTranslator/ReasonerAPI/issues/159
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Reply by patrickkwang** [reply]
Complete! #214

**Comment by edeutsch** [comment]
Enable specification of QEdge "length" (N-M intermediate hops)
- https://github.com/NCATSTranslator/ReasonerAPI/issues/154
- Vote with emoji: heart, thumbs up, nothing, thumbs down

**Comment by edeutsch** [comment]
Multiple Queries (AND/OR them together)
- #105
- Vote with emoji: heart, thumbs up, nothing, thumbs down
